SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1599,6 +1645,46 @@ bool ExtractComponentSetsFromSchemaJson(const FString& SchemaJsonPath, TMap<uint
 			SAFE_TRYGETFIELD((*CompObject), Number, "componentId", ComponentId);
 
 			ComponentMap.Add(ComponentName, ComponentId);
+
+			const TArray<TSharedPtr<FJsonValue>>* FieldArray;
+			SAFE_TRYGETFIELD((*CompObject), Array, "fields", FieldArray);
+
+			uint32 TypeId = OutTypeIdToFieldIdsMap.Num();
+			if (FieldArray->Num() > 0)
+			{
+				SCHEMA_GEN_ERROR(OutComponentIdToTypeIdsMap.Contains(ComponentId),
+								 TEXT("The schema bundle contains duplicate component IDs with component %s."), *ComponentName, TypeId);
+				OutComponentIdToTypeIdsMap.Add(ComponentId, TypeId);
+
+				SCHEMA_GEN_ERROR(OutTypeIdToFieldIdsMap.Contains(TypeId),
+								 TEXT("Ran into a typeId collision with component %s, typeId: %d."), *ComponentName, TypeId);
+				OutTypeIdToFieldIdsMap.Add(TypeId);
+			}
+
+			for (const auto& ArrayValue : *FieldArray)
+			{
+				const TSharedPtr<FJsonObject>* ArrayObject;
+				SAFE_TRYGET(ArrayValue, Object, ArrayObject);
+
+				int32 FieldId;
+				SAFE_TRYGETFIELD((*ArrayObject), Number, "fieldId", FieldId);
+
+				SCHEMA_GEN_ERROR(OutTypeIdToFieldIdsMap[TypeId].FieldIds.Contains(FieldId),
```

Comment:
nit - `OutTypeIdToFieldIdsMap[TypeId]` can be cached when it's first added to the map

Reply:
did something similar